Application/Education/Certificate/Generator/Service/DataCertificate/IDataESBD.php

```diff
@@ -0,0 +1,637 @@
+//            if (!$Data->getCertificateFieldByCertificateAndField($tblCertificate, $FieldName)){
+//                $Data->createCertificateField($tblCertificate, $FieldName, 700);
+//            }
+//            // Begrenzung des Einschätzungfelds
+//            $FieldName = 'Rating';
+//            if (!$Data->getCertificateFieldByCertificateAndField($tblCertificate, $FieldName)){
+//                $Data->createCertificateField($tblCertificate, $FieldName, 600);
+//            }
+        }
+        if ($tblCertificate && !$Data->getCertificateGradeAll($tblCertificate)) {
+            $Data->setCertificateGradeAllStandard($tblCertificate);
+        }
+        if ($tblCertificate && !$Data->getCertificateSubjectAll($tblCertificate)) {
+
+            $Data->setCertificateSubject($tblCertificate, 'DE', 1, 1);
+            $Data->setCertificateSubject($tblCertificate, 'SU', 1, 2);
+            $Data->setCertificateSubject($tblCertificate, 'EN', 1, 3);
+            $Data->setCertificateSubject($tblCertificate, 'KU', 1, 4);
+            $Data->setCertificateSubject($tblCertificate, 'MU', 1, 5);
+
+            $Data->setCertificateSubject($tblCertificate, 'MA', 2, 1);
+            $Data->setCertificateSubject($tblCertificate, 'SPO', 2, 2, true);
+            $Data->setCertificateSubject($tblCertificate, 'RE/e', 2, 3);
+            $Data->setCertificateSubject($tblCertificate, 'WE', 2, 4);
+        }
+    }
+    /**
+     * @param Data        $Data
+     * @param TblConsumer $tblConsumerCertificate
+     */
+    private static function setEsbdGsJOne(Data $Data, TblConsumer $tblConsumerCertificate)
+    {
+
+        $tblCertificate = $Data->createCertificate('Grundschule Jahreszeugnis', 'Klasse 1-3', 'ESBD\EsbdGsJOne', $tblConsumerCertificate);
+        if ($tblCertificate) {
+            if ($Data->getTblSchoolTypePrimary()) {
+                $Data->updateCertificate($tblCertificate, $Data->getTblCertificateTypeYear(), $Data->getTblSchoolTypePrimary());
+                if (!$Data->getCertificateLevelAllByCertificate($tblCertificate)) {
+                    if (($tblLevel = Division::useService()->getLevelBy($Data->getTblSchoolTypePrimary(), '1'))) {
+                        $Data->createCertificateLevel($tblCertificate, $tblLevel);
+                    }
+                    if (($tblLevel = Division::useService()->getLevelBy($Data->getTblSchoolTypePrimary(), '2'))) {
+                        $Data->createCertificateLevel($tblCertificate, $tblLevel);
+                    }
+                    if (($tblLevel = Division::useService()->getLevelBy($Data->getTblSchoolTypePrimary(), '3'))) {
+                        $Data->createCertificateLevel($tblCertificate, $tblLevel);
+                    }
+                }
+            }
+        }
+    }
+    /**
+     * @param Data        $Data
+     * @param TblConsumer $tblConsumerCertificate
+     */
+    private static function setEsbdGymHjInfo(Data $Data, TblConsumer $tblConsumerCertificate)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The method setEsbdGymHjInfo() has an NPath complexity of 603. The configured NPath complexity threshold is 200.](https://app.codacy.com/manual/gerdchristian-kunze/SPHERE-Framework/pullRequest?prid=5646354)